os/kernel/log_dump/log_dump.c

```diff
@@ -109,10 +136,69 @@ int log_dump_init(void)
 	return LOG_DUMP_OK;
 }
 
-void log_dump_read_init(void)
+unsigned char *last_comp_block;
+size_t last_comp_block_size;
+int last_comp_block_ptr;
+bool last_block_compression;
+
+int log_dump_read_wake(void)
+{
+	last_block_compression = true;
+	struct log_dump_msg_s msg;
+	int ret;
+
+	msg.comp = true;
+	msg.ch = 'c';
+	ret = mq_send(log_dump_get_mqfd(), (const char *)&msg, sizeof(struct log_dump_msg_s), 100);
+	if (ret < 0) {
+		lldbg("failed mq_send ret = %d\n", ret);
+		return ERROR;
+	}
+
+	while (last_block_compression) ;
+	return 0;
+}
```

Comment:
I can't get the function activity. Could you explain?

Reply:
This is used to wake up the compression thread to compress current data.